move_group/src/move_group_context.cpp

```diff
@@ -89,3 +89,41 @@ bool move_group::MoveGroupContext::status() const
     return false;
   }
 }
+
+/** Validate that first point of given a trajectory matches current state of robot */
+bool move_group::MoveGroupContext::validateTrajectory(const trajectory_msgs::JointTrajectory &trajectory) const
+{
+  if (trajectory.points.empty())
+    return true;
+
+  const planning_scene_monitor::CurrentStateMonitorPtr csm = planning_scene_monitor_->getStateMonitor();
+  robot_state::RobotStatePtr current_state;
+  if (!csm->waitForCurrentState(1.0) || !(current_state = csm->getCurrentState()))
+  {
+    ROS_WARN("Failed to receive full current joint state");
+    return false;
+  }
```

Comment:
Shouldn't we check `csm->getCurrentStateTime()` and compare to `ros::Time::now()` here to see if our state is up to date?